qiskit/pulse/commands/parametric_pulses.py

```diff
@@ -0,0 +1,292 @@
+                               zero_ends=False)
+
+
+class Drag(ParametricPulse):
+    """
+    A pulse whose envelope is shaped by a drag pulse. This is so named by the technique
+    Derivative Removal by Adiabatic Gate (DRAG). It is constructed to reduce the frequency
+    spectrum of a normal gaussian curve near the |1>-|2> transition.
+
+        f(x) = Gaussian + 1j * beta * d/dx [Gaussian]
+             = Gaussian + 1j * beta * (-x / sigma) [Gaussian]
+
+    where 'Gaussian' is:
+        Gaussian(x, amp, sigma) = amp * exp( -(1/2) * (x - duration/2)^2 / sigma^2) )
+
+    Ref:
+        [1] Gambetta, J. M., Motzoi, F., Merkel, S. T. & Wilhelm, F. K.
+        Analytic control methods for high-fidelity unitary operations
+        in a weakly nonlinear oscillator. Phys. Rev. A 83, 012308 (2011).
+        [2] F. Motzoi, J. M. Gambetta, P. Rebentrost, and F. K. Wilhelm
+        Phys. Rev. Lett. 103, 110501 – Published 8 September 2009
+    """
+
+    def __init__(self,
+                 duration: int,
+                 amp: complex,
+                 sigma: float,
+                 beta: int):
+        """Initialize the drag command.
+
+        Args:
+            duration: Pulse length in terms of the the sampling period `dt`.
+            amp: The amplitude of the Drag envelope.
+            sigma: A measure of how wide or narrow the Gaussian peak is; described mathematically
+                   in the class docstring.
+            beta: The correction amplitude.
+        """
+        validate_params(amp=amp, sigma=sigma, beta=beta)
+        self._amp = complex(amp)
+        self._sigma = sigma
+        self._beta = beta
+        super().__init__(duration=duration)
+
+    @property
+    def amp(self):
+        return self._amp
+
+    @property
+    def sigma(self):
+        return self._sigma
+
+    @property
+    def beta(self):
+        return self._beta
+
+    def get_sample_pulse(self) -> SamplePulse:
+        return drag(duration=self.duration, amp=self.amp, sigma=self.sigma,
+                    beta=self.beta, zero_ends=False)
+
+
+class ConstantPulse(ParametricPulse):
+    """
+    A simple constant pulse, with an amplitude value and a duration:
+
+        f(x) = amp    ,  0 <= x < duration
+        f(x) = 0      ,  elsewhere
+    """
+
+    def __init__(self,
+                 duration: int,
+                 amp: complex):
+        """
+        Initialize the constant valued pulse command.
+
+        Args:
+            duration: Pulse length in terms of the the sampling period `dt`.
+            amp: The amplitude of the constant square pulse.
+        """
+        validate_params(amp=amp)
+        self._amp = complex(amp)
+        super().__init__(duration=duration)
+
+    @property
+    def amp(self):
+        return self._amp
+
+    def get_sample_pulse(self) -> SamplePulse:
+        return constant(duration=self.duration, amp=self.amp)
+
+
+def validate_params(**kwargs):
```

Comment:
Another possible case is provider A and B prepare the same `Gaussian` parametric, for example, but parameter has different meaning. In this case, how to switch validation?

Reply:
Yeah, you guys are right, I was just being lazy because right now the validation for all the classes is actually the same. I'll have to update this. Sometimes it really is always the same, like for `amp`, which is constrained to norm 1 across all of open pulse
@nkanazawa1989 validation which depends on provider A vs B would not go here, but rather in a `validation` module which we have yet to implement. Any validation which requires constraints provided by backends will have to go there. When that is done, the user would still get an error before executing their code, but it would occur at assemble time rather than build time